### DISCUSSION TITLE
Minor release 1.8.0, now with RP2350 support

### DISCUSSION THREAD

**Discussion by tabemann** [tabemann/zeptoforth#145] [General]
This is the first non-pre-release release of zeptoforth with RP2350 support. Included in RP2350 support it adds support for 48 GPIO's (note that only 30 GPIO's, the same number as on the RP2040, are accessible on the RP2350A), a third PIO peripheral (with four extra state machines), four more PWM peripherals (with two channels per peripheral; note that only eight PWM peripherals, the same number as on the RP2040, are accessible on the RP2350A), a second hardware timer peripheral, the Always-On Timer (which is used to emulate the RTC provided by the RP2040, but with added support for millisecond resolution and multiple clock sources), and four more DMA channels relative to the feature set provided by the RP2040. It also adds optional support for PSRAM; note that this is only usable on RP2350 boards that include PSRAM, such as the Pimoroni Pico Plus 2. Note that HSTX is not currently supported, but may be supported in a future release.

Also note that there are a number of minor bugfixes and additions to things other than RP2350 support, such as fixing bugs in zeptoed and improving the DMA API on the RP2040 in addition to the API exposed on the RP2350.

Note that there are no separate "non-big" and "big" builds for the RP2350; all RP2350 builds allot 4 KiB to the bootloader, 36 KiB to the kernel, 1912 KiB to the flash dictionary, 92 KiB to the flash dictionary index (the "mini-dictionary"), and 2 MiB to block/FAT32 filesystem storage.

<hr /><em>This discussion was created from the release <a href='https://github.com/tabemann/zeptoforth/releases/tag/v1.8.0'>Minor release 1.8.0, now with RP2350 support</a>.</em>

**Comment by znmeb** [comment]
RP2350 support is hardly minor. Thanks!!

**Comment by tabemann** [comment]
It's "minor" because I am reserving the next "major" release, 2.0.0, for
whenever I get around to rewriting the zeptoforth code generator to be
register-assigning.
